src/ResourceManager/Network/Commands.Network/VirtualNetworkGatewayConnection/NewAzureVirtualNetworkGatewayConnectionCommand.cs

```diff
@@ -130,6 +131,18 @@ public class NewAzureVirtualNetworkGatewayConnectionCommand : VirtualNetworkGate
             HelpMessage = "Do not ask for confirmation if you want to overrite a resource")]
         public SwitchParameter Force { get; set; }
 
+        [Parameter(
+            Mandatory = false,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = "Whether to enable policy-based traffic selectors for a S2S connection")]
+        public string UsePolicyBasedTrafficSelectors { get; set; }
```

Comment:
@henry416 why use a `string` here if you're going to just parse it into a `bool`? The recommended pattern is to add a parameter called `EnablePolicyBasedTrafficSelectors`, and when this switch is enabled, users can provide a value for the `IpsecPolicies` parameter. These two parameters would need to be in a separate parameter set from the default one, and `EnablePolicyBasedTrafficSelectors` would need to be a mandatory parameter. Let me know if you have any questions about this.